up_client_socket/rust/utransport-socket/src/lib.rs

```diff
@@ -0,0 +1,420 @@
+         dbg!("check_on_receive..\n");
+         match listeners {
+             Entry::Vacant(_) => {
+                 dbg!("check_on_receive 1..\n");
+                 return Err(UStatus::fail_with_code(
+                     UCode::NOT_FOUND,
+                     format!("No listeners registered for topic: {:?}", &uuri),
+                 ));
+             }
+             Entry::Occupied(mut e) => {
+                 let occupied = e.get_mut();
+                 dbg!("check_on_receive 2..\n");
+                 if occupied.is_empty() {
+                     return Err(UStatus::fail_with_code(
+                         UCode::NOT_FOUND,
+                         format!("No listeners registered for topic: {:?}", &uuri),
+                     ));
+                 }
+                 dbg!("invoking listner on receive..\n");
+                 for listener in occupied.iter() {
+                     let task_listener = listener.clone();
+                     let task_umessage = umessage.clone();
+                     dbg!("invoking listner on receive\n");
+                     task::spawn(async move { task_listener.on_receive(task_umessage).await });
+                 }
+             }
+         }
+ 
+         Ok(())
+     }
+ }
+ 
+ #[async_trait]
+ impl UTransport for UTransportSocket {
+     /// Sends a message using this transport's message exchange mechanism.
+     ///
+     /// # Arguments
+     ///
+     /// * `message` - The message to send. The `type`, `source` and`sink` properties of the [`crate::UAttributes`] contained
+     ///   in the message determine the addressing semantics:
+     ///   * `source` - The origin of the message being sent. The address must be resolved. The semantics of the address
+     ///     depends on the value of the given [attributes' type](crate::UAttributes::type_) property .
+     ///     * For a [`PUBLISH`](crate::UMessageType::UMESSAGE_TYPE_PUBLISH) message, this is the topic that the message should be published to,
+     ///     * for a [`REQUEST`](crate::UMessageType::UMESSAGE_TYPE_REQUEST) message, this is the *reply-to* address that the sender expects to receive the response at, and
+     ///     * for a [`RESPONSE`](crate::UMessageType::UMESSAGE_TYPE_RESPONSE) message, this identifies the method that has been invoked.
+     ///   * `sink` - For a `notification`, an RPC `request` or RPC `response` message, the (resolved) address that the message
+     ///     should be sent to.
+     ///
+     /// # Errors
+     ///
+     /// Returns an error if the message could not be sent.
+     async fn send(&self, message: UMessage) -> Result<(), UStatus> {
+         let mut socket_clone = self.socket_sync.try_clone().expect("issue in sending data");
+ 
+         let umsg_serialized = message
+             .clone()
+             .write_to_bytes()
+             .expect("Send Serialization Issue");
+         let _result = UMessage::parse_from_bytes(&umsg_serialized.clone());
```

Comment:
Why is the message serialized and then deserialized but the deserialized result not used?

Reply:
residue from previous implementation. removed.